src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt

```diff
@@ -190,26 +189,38 @@ private class MembersGenerator(
     val itemsToImport: Set<ImportCandidate> get() = renderer.itemsToImport
 
     fun createTraitMembers(members: Collection<RsAbstractable>): RsMembers {
-        val subst = trait.subst
         val body = members.joinToString(separator = "\n", transform = {
-            when (it) {
-                is RsConstant -> {
-                    val initialValue = RsDefaultValueBuilder(it.knownItems, it.containingMod, factory, true)
-                        .buildFor(it.typeReference?.type?.substitute(subst) ?: TyUnknown, emptyMap())
-                    "    const ${it.nameLikeElement.text}: ${it.typeReference?.renderTypeReference() ?: "_"} = ${initialValue.text};"
-                }
-                is RsTypeAlias ->
-                    "    type ${it.escapedName} = ();"
-                is RsFunction ->
-                    "    ${it.renderFunctionSignature()}{\n        todo!()\n    }"
-                else ->
-                    error("Unknown trait member")
-            }
+            "    ${renderAbstractable(it)}"
         })
 
         return factory.createMembers(body)
     }
 
+    fun renderAbstractable(element: RsAbstractable): String {
+        val subst = trait.subst
+        return when (element) {
+            is RsConstant -> {
+                val initialValue = RsDefaultValueBuilder(element.knownItems, element.containingMod, factory, true)
+                    .buildFor(element.typeReference?.type?.substitute(subst) ?: TyUnknown, emptyMap())
+                "const ${element.nameLikeElement.text}: ${element.typeReference?.renderTypeReference() ?: "_"} = ${initialValue.text};"
+            }
+            is RsTypeAlias ->
+                "type ${element.escapedName} = ();"
+            is RsFunction ->
+                "${element.renderFunctionSignature().withSpace()}{\n        todo!()\n    }"
+            else ->
+                error("Unknown trait member")
+        }
+    }
+
     private fun RsFunction.renderFunctionSignature(): String = renderer.renderFunctionSignature(this)
     private fun RsTypeReference.renderTypeReference(): String = renderer.renderTypeReference(this)
 }
+
+private fun String.withSpace(): String {
```

Comment:
Hmm, when is it possible that `renderFunctionSignature` ends with space?

Reply:
Oh, it's not. I was just surprised that the space was added automatically in implement members fix, but not here in completion. Probably in implement members the auto formatter is run automatically.